utilities/backupable/backupable_db.cc

```diff
@@ -805,10 +809,11 @@ Status BackupEngineImpl::Initialize() {
           GetAbsolutePath(GetPrivateFileRel(backup_iter->first)), backup_env_,
           &abs_path_to_size);
       if (s.ok()) {
-        s = backup_iter->second->LoadFromFile(options_.backup_dir,
-                                              abs_path_to_size);
+        s = backup_iter->second->LoadFromFile(
+            options_.backup_dir, abs_path_to_size, options_.info_log,
+            &reported_ignored_fields_);
       }
-      if (s.IsCorruption()) {
+      if (s.IsCorruption() || s.IsNotSupported()) {
```

Comment:
That is a very good point. In theory we could try to heuristically determine the files associated with a backup even when we know we cannot fully faithfully process the meta file, but that seems fragile and annoying to test.

The good news is that we only auto-GC if/when a write operation on BackupEngine is called. By implication, by invoking a write operation, you are essentially agreeing that you are OK with the set of good backups and will not be attempting some intervention for recovery of corrupted backups. Technically, this is already true, but we should probably re-emphasize this when putting schema update into production, because we will increase the likelihood of hitting this.